dags/synaptor_ops.py

```diff
@@ -21,13 +21,145 @@
 TASK_QUEUE_NAME = "synaptor"
 
 
-# Python callables (for PythonOperators)
-def generate_ngl_link_op() -> None:
+# Op functions
+def generate_ngl_link(
+    net_output_path: str,
+    seg_path: str,
+    workflowtype: str,
+    storagedir: str,
+    img_path: Optional[str] = None,
+    voxelres: Optional[tuple[int, int, int]] = None,
+) -> None:
     """Generates a neuroglancer link to view the results."""
-    pass
+    layers = [
+        ImageLayer("network output", net_output_path),
+        SegLayer("synaptor segmentation", seg_path),
+    ]
+
+    if img_path:
+        layers = [ImageLayer("image", img_path)] + layers
+
+    payload = generate_ng_payload(layers)
+
+    if "Assignment" in workflowtype:
+        presyn_pts, postsyn_pts = read_pts(storagedir)
+        payload = add_annotation_layer(payload, presyn_pts, postsyn_pts, voxelres)
+
+    basename = dump_payload(payload, storagedir)
+
+    slack_message(wrap_payload(os.path.join(storagedir, basename)), broadcast=True)
+
+
+def dump_payload(payload: dict, storagedir: str) -> str:
+    from cloudfiles import CloudFiles
+    basename = "ng.json"
+
+    cf = CloudFiles(storagedir)
+    cf.put_json(basename, payload)
+
+    return basename
+
+
+def read_pts(storagedir: str) -> tuple[list, list]:
+    from cloudfiles import CloudFiles
+
+    cf = CloudFiles(storagedir)
+    maybe_content = cf.get("final_edgelist.df")
```

Comment:
Similar to the previous comment, I guess you need to add a function to mount the secrets and get files from some bucket. Maybe it is a good idea to try to run this in a local deployment, that would make all the secret issues manifest.

Reply:
I don't understand what secret issues you're referring to here. Are you worried about when the manager node doesn't have default credential access to the required bucket or something?